receiver/githubactionsreceiver/trace_receiver.go

```diff
@@ -0,0 +1,166 @@
+	params receiver.CreateSettings,
+	config *Config,
+	nextConsumer consumer.Traces,
+) (*githubActionsReceiver, error) {
+	if config.Endpoint == "" {
+		return nil, errMissingEndpoint
+	}
+
+	transport := "http"
+	if config.TLSSetting != nil {
+		transport = "https"
+	}
+
+	obsrecv, err := receiverhelper.NewObsReport(receiverhelper.ObsReportSettings{
+		ReceiverID:             params.ID,
+		Transport:              transport,
+		ReceiverCreateSettings: params,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	client := github.NewClient(nil)
+
+	gar := &githubActionsReceiver{
+		nextConsumer:   nextConsumer,
+		config:         config,
+		createSettings: params,
+		logger:         params.Logger,
+		obsrecv:        obsrecv,
+		ghClient:       client,
+	}
+
+	return gar, nil
+}
+
+func (gar *githubActionsReceiver) Start(ctx context.Context, host component.Host) error {
+	endpoint := fmt.Sprintf("%s%s", gar.config.Endpoint, gar.config.Path)
+	gar.logger.Info("Starting GithubActions server", zap.String("endpoint", endpoint))
+	gar.server = &http.Server{
+		Addr:    gar.config.ServerConfig.Endpoint,
+		Handler: gar,
+	}
+
+	gar.shutdownWG.Add(1)
+	go func() {
+		defer gar.shutdownWG.Done()
+
+		if errHTTP := gar.server.ListenAndServe(); !errors.Is(errHTTP, http.ErrServerClosed) && errHTTP != nil {
+			gar.createSettings.TelemetrySettings.Logger.Error("Server closed with error", zap.Error(errHTTP))
+		}
+	}()
+
+	return nil
+}
+
+func (gar *githubActionsReceiver) Shutdown(ctx context.Context) error {
+	var err error
+	if gar.server != nil {
+		err = gar.server.Close()
+	}
+	gar.shutdownWG.Wait()
+	return err
+}
+
+func (gar *githubActionsReceiver) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	if r.URL.Path != gar.config.Path {
+		http.Error(w, "Not found", http.StatusNotFound)
+		return
+	}
+
+	payload, err := github.ValidatePayload(r, []byte(gar.config.Secret))
+	if err != nil {
+		gar.logger.Debug("Payload validation failed", zap.Error(err))
+		http.Error(w, "Invalid payload or signature", http.StatusBadRequest)
+		return
+	}
+
+	event, err := github.ParseWebHook(github.WebHookType(r), payload)
+	if err != nil {
+		gar.logger.Debug("Webhook parsing failed", zap.Error(err))
+		http.Error(w, "Failed to parse webhook", http.StatusBadRequest)
+		return
+	}
+
+	gar.logger.Debug("Received GitHub event", zap.Any("event", event))
+
+	// Filter events based on the status
+	switch e := event.(type) {
+	case *github.WorkflowJobEvent:
+		if e.GetWorkflowJob().GetStatus() != "completed" {
+			gar.logger.Debug("Skipping non-completed WorkflowJobEvent", zap.String("status", e.GetWorkflowJob().GetStatus()))
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+	case *github.WorkflowRunEvent:
+		if e.GetWorkflowRun().GetStatus() != "completed" {
+			gar.logger.Debug("Skipping non-completed WorkflowRunEvent", zap.String("status", e.GetWorkflowRun().GetStatus()))
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
```

Comment:
```suggestion
		}
	default:
		gar.logger.Debug("Skipping unsupported event type", zap.String("event_type", github.WebHookType(r)))
		w.WriteHeader(http.StatusNoContent)
		return
```

The GitHub App I'm using has `checks: write` permissions. This causes the webhook to automatically subscribe to the `[check_run](https://docs.github.com/en/webhooks/webhook-events-and-payloads#check_run)` and `[check_suite](https://docs.github.com/en/webhooks/webhook-events-and-payloads#check_suite)` events.

Because of that, I was getting errors in the `eventToTraces` function, which could not process the event.

Generally, this suggestion would skip all events other than `workflow_run` and `workflow_job`.

WDYT?

Reply:
Good catch and it does make sense not to process anything but `workflow_run` and `workflow_job` events. 

Instead of passing the event to to `eventToTraces`, I'd prefer to filter these events prior to passing to that func. I'l refactor accordingly.

Reply:
Just pushed the new changes. It'll filter out any non events which are not of type `workflow_run` and `workflow_job`.

Reply:
Thanks @krzko!